File: java/1.4 Iteradores y Colecciones de Objetos/1.4.3 Colecciones/Ejemplo01/com/ditamex/main/VectorTest.java
package com.ditamex.main;

import java.util.Vector;

public class VectorTest
{
    public static void main(String args[]) {
        /* 
         * Revisar la documentación de la clase Vector
         * https://docs.oracle.com/javase/8/docs/api/java/util/Vector.html
         */
        
        //Se inicializa el vector con "cero" elementos
        Vector colores = new Vector();
        
        //Se agregan valores iniciales al vector
        colores.add("Rojo");
        colores.add("Blanco");
        colores.add("Amarillo");
        colores.add("Negro");
        colores.add("Azul"); 
                
        System.out.println("Colores: " + colores);        
        
        //Primer operación "add"
        colores.add("Rosa");
        System.out.println("Colores: " + colores);

        //Segunda operación "set"
        colores.set(3, "Verde");
        System.out.println("Colores: " + colores);        
                
        //Tercera operación "remove"
        colores.remove(1);
        System.out.println("Colores: " + colores);  
        
        //Cuarta operación "get"
        String color = (String) colores.get(4);
        System.out.println("Color: " + color);
    }
}

File: c++/1.4 Iteradores y Colecciones de Objetos/1.4.3 Colecciones/Ejemplo01/VectorTest.cpp
#include <iostream>
#include <vector>

using namespace std;

void imprimeVector(string nombre, vector<string> miVector) {
  cout << nombre << ": [";
  for (int x = 0; x < miVector.size(); x++) {
    cout << miVector[x];
    if (x != miVector.size() - 1) {
      cout << ", ";
    }
  }
  cout << "]" << endl;
}

int main(int argc, char const *argv[]) {
  // Se inicializa el vector con "cero" elementos
  vector<string> colores;

  // Se agregan valores iniciales al vector
  colores.push_back("Rojo");
  colores.push_back("Blanco");
  colores.push_back("Amarillo");
  colores.push_back("Negro");
  colores.push_back("Azul");

  imprimeVector("Colores", colores);

  // Primer operación "add: push_back()"
  colores.push_back("Rosa");
  imprimeVector("Colores", colores);

  // Segunda operación "set: at()"
  colores.at(3) = "Verde";  // también: colores[3] = "Verde";
  imprimeVector("Colores", colores);

  // Tercera operación "remove: erase()"
  colores.erase(colores.begin() + 1);
  imprimeVector("Colores", colores);

  // Cuarta operación "get: at()"
  string color = colores.at(4);  // también: string color = colores[4]
  cout << "Color: " << color << endl;

  return 0;
}
